compute/cloud-client/src/main/java/compute/disks/consistencygroup/CloneDisksFromConsistencyGroup.java

```diff
@@ -0,0 +1,92 @@
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class CloneDisksFromConsistencyGroup {
+
+  public static void main(String[] args)
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    // TODO(developer): Replace these variables before running the sample.
+    // Project ID or project number of the Cloud project you want to use.
+    String project = "YOUR_PROJECT_ID";
+    // Name of the region or zone in which your disk is located.
+    String disksLocation = "us-central1";
+    // Name of the consistency group you want to clone disks from.
+    String consistencyGroupName = "YOUR_CONSISTENCY_GROUP_NAME";
+    // Name of the region in which your consistency group is located.
+    String consistencyGroupLocation = "us-central1";
+
+    cloneDisksFromConsistencyGroup(
+            project, disksLocation, consistencyGroupName, consistencyGroupLocation);
+  }
+
+  // Clones disks from a consistency group.
+  public static Operation.Status cloneDisksFromConsistencyGroup(String project,
+      String disksLocation, String consistencyGroupName, String consistencyGroupLocation)
+      throws IOException, InterruptedException, ExecutionException, TimeoutException {
+    String sourceConsistencyGroupPolicy = String.format(
+            "projects/%s/regions/%s/resourcePolicies/%s", project, consistencyGroupLocation,
+            consistencyGroupName);
+    Operation response;
+    if (Character.isDigit(disksLocation.charAt(disksLocation.length() - 1))) {
+      // Initialize client that will be used to send requests. This client only needs to be created
+      // once, and can be reused for multiple requests.
+      try (RegionDisksClient disksClient = RegionDisksClient.create()) {
+        BulkInsertRegionDiskRequest request = BulkInsertRegionDiskRequest.newBuilder()
+                .setProject(project)
+                .setRegion(disksLocation)
+                .setBulkInsertDiskResourceResource(
+                        BulkInsertDiskResource.newBuilder()
+                                .setSourceConsistencyGroupPolicy(sourceConsistencyGroupPolicy)
+                                .build())
+                .build();
+
+        response = disksClient.bulkInsertAsync(request).get(3, TimeUnit.MINUTES);
+      }
+    } else {
+      try (DisksClient disksClient = DisksClient.create()) {
+        BulkInsertDiskRequest request = BulkInsertDiskRequest.newBuilder()
+                .setProject(project)
+                .setZone(disksLocation)
+                .setBulkInsertDiskResourceResource(
+                        BulkInsertDiskResource.newBuilder()
+                                .setSourceConsistencyGroupPolicy(sourceConsistencyGroupPolicy)
+                                .build())
+                .build();
+
+        response = disksClient.bulkInsertAsync(request).get(3, TimeUnit.MINUTES);
+      }
+    }
```

Comment:
this code does two code samples in one. please, fix this either by demonstrating cloning on a zonal disk or creating two code samples for different demos. I recommend to reach out to tech writer to see if it makes sense to have two samples.
Note that the current documentation use of pseudo "LOCATION" parameter that is further have to be edited by a user is not a recommended practice.

Reply:
Thank you! Created sample for zonal location